### DISCUSSION TITLE
help with reset breakpoint and watchdog implications

### DISCUSSION THREAD

**Discussion by PatrickTague** [blackmagic-debug/blackmagic#1695] [Q&A] [answered]
hi all. i'm using bmps for an embedded course where students essentially build their own rtos from bare metal on a board with an nRF52840.  we're using largely stock firmware, with the exception of enabling RTT and trimming out some code to make the image small enough to flash.

i noticed that when we program in a system reset ("soft reset"), the bmp catches it and triggers a breakpoint.  this didn't happen with our previous debugger, and i'd like to eliminate it if possible.  does anyone know how i could quickly modify the blackmagic firmware to not trigger this breakpoint on reset?  it's particularly problematic if students build a watchdog.

**Comment by dragonmux** [comment, marked as answer]
This is due to the vector catch system for Cortex-M cores - the default is to catch all, which is normally desired behaviour, however if you need to disable a specific core interrupt vector's interception, then the [`monitor vector_catch`](https://github.com/blackmagic-debug/blackmagic/blob/5df46dc891dc07f18671fe9b8a9f299652f6fd97/src/target/cortexm.c#L1388) command exists to configure and help with this:

`monitor vector_catch disable reset` to disable catch and interception of the reset vector

**Reply by PatrickTague** [reply]
oh, thank you!  very nice.
